content/en/docs/reference/access-authn-authz/admission-controllers.md

```diff
@@ -200,9 +200,11 @@ and automatically adds a default storage class to them.
 This way, users that do not request any special storage class do not need to care about them at all and they
 will get the default one.
 
-This admission controller does not do anything when no default storage class is configured. When more than one storage
-class is marked as default, it rejects any creation of `PersistentVolumeClaim` with an error and an administrator
-must revisit their `StorageClass` objects and mark only one as default.
+This admission controller does nothing when no default `StorageClass` exists. When more than one storage
+class is marked as default, and you then create a `PersistentVolumeClaim` with no `storageClassName` set, 
+Kubernetes uses the most recently created default `StorageClass`.
+When a `PersistentVolumeClaim` is created with a specified `volumeName`, it remains in a pending state 
+if the static volume's `storageClassName` does not match the `StorageClass` on the `PersistentVolumeClaim`.
```

Comment:
What if I'm creating a PVC with 'volumeName' specified, but no 'storageClassName'?
The PVC still has to be checked for storage class matching?

Reply:
Yes, the PVC still checks for matching storage class, the below show the events on a PVC with a specified `volumeName`  that has been mutated by the `DefaultStroageClass` admission controller.

```
controlplane $ k describe pvc pvc-nginx 
Name:          pvc-nginx
Namespace:     default
StorageClass:  nfs-csi
Status:        Pending
Volume:        pv-nginx
Labels:        <none>
Annotations:   <none>
Finalizers:    [kubernetes.io/pvc-protection]
Capacity:      0
Access Modes:  
VolumeMode:    Filesystem
Used By:       nginx-nfs-example
Events:
  Type     Reason          Age               From                         Message
  ----     ------          ----              ----                         -------
  Warning  VolumeMismatch  3s (x3 over 23s)  persistentvolume-controller  Cannot bind to requested volume "pv-nginx": storageClassName does not match
```

Reply:
I mean ... remove the 'storageClassName' from the PVC, setting only 'volumeName'.
Will this binding succeed? I guess so.
The admission controller has nothing to check for this PVC, right?

Reply:
The admission controller may backfill the storage class for the PVC when the binding is a success. But that is a different topic. The PVC will not be pending for storage class matching.

Reply:
I can't remove the 'storageClassName' from the PVC, setting only 'volumeName', the spec is immutable after creation.
As long as here is atleast one storage class setup as default, here is also no way to create a PVC without a `storageClass`, the admission controller is always on the look out for PVC without one specified to mutate it.

Reply:
> I can't remove the `storageClass` from the PVC, setting only `volumeName`, the spec is immutable after creation.

Immutable should be fine.

> As long as here is at least one storage class setup as default, here is also no way to create a PVC without a `storageClass`, the admission controller is always on the look out for PVC without one specified to mutate it.

This is a surprise for me. Maybe we need to consult SIG Storage for this.

Reply:
The question is this, @msau42:

If I am creating a PVC in a cluster, and the cluster has a default storage class.
I'm not specifying `storageClassName` in my PVC spec. I'm only specifying `volumeName`.
Can the PVC be created?

Right. I can understand that the PVC may and may not receive a `storageClassName` after creation, but that is a different topic.